sympy/matrices/common.py

```diff
@@ -2420,19 +2437,35 @@ def pow(self, exp, jordan=None, dotprodsimp=None):
             # When certain conditions are met,
             # Jordan block algorithm is faster than
             # computation by recursion.
-            elif jordan_pow is not None and (jordan or \
-                    (jordan is not False and a.rows == 2 and exp > 100000)):
+            elif method == 'jordan':
                 try:
                     return jordan_pow(exp)
                 except MatrixError:
-                    if jordan:
+                    if method == 'jordan':
                         raise
 
-            if _get_intermediate_simp_bool(True, dotprodsimp):
+            if method == 'cayley':
+                return a._eval_pow_by_cayley(exp)
```

Comment:
These things should outside the enclosing if

Reply:
Done. Is this okay?